core/src/main/java/org/apache/accumulo/core/metadata/schema/TabletMetadata.java

```diff
@@ -202,7 +206,74 @@ public enum ColumnType {
     COMPACTED,
     USER_COMPACTION_REQUESTED,
     UNSPLITTABLE,
-    MERGEABILITY
+    MERGEABILITY;
+
+    public static Set<ByteSequence> resolveFamilies(Set<ColumnType> columns) {
```

Comment:
There is a similar method in `TabletsMetadata.Builder.fetch`.  I wonder if the logic in that method to decode ColumnTypes to families and qualifiers could be pulled out and re-used.

Reply:
That's a good idea, I'll take a look at refactoring it. One nice benefit of consolidation is that if we add a new column family it reduces the chance of it being forgotten to be handled

Reply:
> The methods are a bit different so we may need to do something like have some immutable maps that map between ColumnType and ColumnFQ or families. 

That would be nice if instead of the switch stmt we had a static `Map<ColumnType, List<Family>>` that was populated in a static code block.  Looked for a way to automatically construct this in a static code block and do not see a way w/ the current code.

Reply:
The only issue I see is that for fetch(), for some ColumnTypes we fetch families and for some we fetch columns. So I think we may need to split the columns into two groups to grab the type we need which should be doable. I assume we don't want to load the entire family if we only care about one column so we still need to keep this.

i'm working on changes for this now.

Reply:
yeah, definitely want to keep the qualifier aspect for code that cares about it.

Reply:
Should this be a follow on issue or should be it be done as part of this PR?

Reply:
Nevermind, I think it probably makes sense to just do the mapping/enum changes as part of this PR vs a follow on as it's easier I think to just do it all at once.